CenterlineMetrics/CenterlineMetrics.py

```diff
@@ -382,11 +443,91 @@ def getExtremeDiameterPoint(self, boolMaximum):
             break
     return target
 
-  def getUnitNodeDisplayString(self, value):
+  def getUnitNodeDisplayString(self, value, category):
     selectionNode = slicer.mrmlScene.GetNodeByID("vtkMRMLSelectionNodeSingleton")
-    unitNode = slicer.mrmlScene.GetNodeByID(selectionNode.GetUnitNodeID("length"))
+    unitNode = slicer.mrmlScene.GetNodeByID(selectionNode.GetUnitNodeID(category))
     return unitNode.GetDisplayStringFromValue(value)
     
+  
+  def createCrossSection(self, center, normal):
+    if self.segmentationNode is None or self.currentSegmentID == "":
+        # Don't print any message, is perhaps intended.
+        return
+    
+    util = Utility()
+    util.islandModelNode = self.islandModelNode
+    
+    # Place a plane perpendicular to the centerline
+    plane = vtk.vtkPlane()
+    plane.SetOrigin(center)
+    plane.SetNormal(normal)
+    
+    # Work on the segment's closed surface
+    closedSurfacePolyData = vtk.vtkPolyData()
+    self.segmentationNode.GetClosedSurfaceRepresentation(self.currentSegmentID, closedSurfacePolyData)
```

Comment:
The closed surface representation must be transformed to the world coordinate system. You can get the transform using `GetTransformBetweenNodes` see complete example in CombineModels.py.

Reply:
> The closed surface representation must be transformed to the world coordinate system. You can get the transform using `GetTransformBetweenNodes`

Ok, one thing at a time.
While I can broadly see the problem in the other changes to make, here I'm clueless.

In fact, the "[World coordinate system](https://www.slicer.org/wiki/Coordinate_systems#World_coordinate_system)" is quite obscure to me. I suppose it's the normal XY axis with (0,0) origin being the lower left corner.
Is it self.segmentationNode that should be transformed to world coordinates ? Really, I'm clueless how to use GetTransformBetweenNodes to do that.

And why should we do that?

Thanks for your replies.

Reply:
"World" coordinate system, also known as "Renderer" or "RAS" coordinate system is the space where nodes are, if no transforms are applied to them. If you apply a transform to a node (e.g., Curved Planar Reformat module computes a straightening transform) then it does not change the point coordinates in a transformed model, just uses the transform to translate/rotate/warp the model for display. Some more information is provided in [Transforms module documentation](https://slicer.readthedocs.io/en/latest/user_guide/modules/transforms.html), but if anything is not clear then let me know and I'll either clarify here or add some more details to the documentation.

The module must not return incorrect results if a parent transform is set to the centerline and/or the segmentation. Ideally, you would apply the transform to the nodes to the world coordinate system, but you may postpone this for a while and for now just report an error if a parent transform set and instruct the user to harden or remove the parent transforms.

Reply:
> The module must not return incorrect results if a parent transform is set to the centerline and/or the segmentation

A segmentation transform is now applied to the model. I suppose that if the user applies different transforms to the segmentation and the centerline, he knows what he is doing.

Reply:
We cannot assume that the user knows what he is doing. User expectation is that things are foolproof.

You can also easily transform the centerline model (https://github.com/PerkLab/SlicerSandbox/blob/db2fea8b59bfc30209740c546892c998f3adb169/CombineModels/CombineModels.py#L324-L329) and either use VTK API to get the coordinates from the transformed polydata or get the coordinate array as a numpy array (https://github.com/Slicer/Slicer/blob/5f6800e7de2ddc917ad820f63daf60a4a6ea65e2/Base/Python/slicer/util.py#L1474-L1476).

Reply:
> User expectation is that things are foolproof.

That's a big problem : the user moves the segmentation to the left, and moves the centerline to the left plus a rotation - what are we supposed to do ? IMO, users are responsible for what they do.

Reply:
I just mean that we need to need to do all the measurements on what we display. Ability to display the triangulated cross-sections will make it absolutely clear what surfaces are measured.